### PR TITLE
Added Parameter Object With Incorrect Ref query for OpenAPI #2854

### LINKED ISSUE #2854: Add Parameter Object With Incorrect Ref query for OpenAPI
### Platform
OpenAPI 3.0

### Description
Parameter Object reference must always point to '#components/parameters'

### PR DESCRIPTION
**Pull request by joaoReigota1**
Signed-off-by: João Reigota <joao.reigota@checkmarx.com>

Closes #2854

**Proposed Changes**
-Added Parameter Object With Incorrect Ref query for OpenAPI 

I submit this contribution under Apache-2.0 license.

### CODE REVIEW

File: assets/queries/openAPI/parameter_object_incorrect_ref/query.rego [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package Cx
+
+import data.generic.openapi as openapi_lib
+
+CxPolicy[result] {
+	doc := input.document[i]
+	openapi_lib.check_openapi(doc) != "undefined"
+	params := doc.paths[name].parameters[n]
```
Comment (rafaela-soares) [inline assets/queries/openAPI/parameter_object_incorrect_ref/query.rego:8]: Parameters can be defined in other attributes - https://swagger.io/docs/specification/describing-parameters/#path-parameters. Will it be relevant to consider?

File: assets/queries/openAPI/parameter_object_incorrect_ref/metadata.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "id": "d40f27e6-15fb-4b56-90f8-fc0ff0291c51",
+  "queryName": "Parameter Object With Incorrect Ref",
```
Comment (rafaela-soares) [inline assets/queries/openAPI/parameter_object_incorrect_ref/metadata.json:3]: The folder name is slightly different from the 'queryName' in the metadata file

Comment (rogeriopeixotocx) [reply assets/queries/openAPI/parameter_object_incorrect_ref/metadata.json:3]: I think it's ok

**Review by felipe-avelar — APPROVED** [review]
LGTM

**Pull request merged**